var/spack/repos/builtin/packages/memsurfer/package.py

```diff
@@ -12,32 +12,31 @@ class Memsurfer(PythonPackage):
        wide variety of large-scale molecular simulations."""
 
     homepage = "https://github.com/LLNL/MemSurfer"
-    git      = "git@github.com:LLNL/MemSurfer.git"
-    # url      = "https://github.com/LLNL/MemSurfer/archive/1.0.tar.gz"
+    git = "git@github.com:LLNL/MemSurfer.git"
 
-    # version('1.0', sha256='06e06eba88754b0c073f1c770981f7bdd501082986e4fbe28399be23b50138de')
-    version('1.0', tag='v1.0', submodules=True)
-    version('master', branch='master', submodules=True)
-    # version('test', branch='ppoisson', submodules=True)
+    version('1.0',     tag='v1.0',       submodules=True)
+    version('master',  branch='master',  submodules=True)
+    version('develop', branch='develop', submodules=True)
 
-    variant('vtkmesa', default=False, description='Enable OSMesa support for VTK')
+    variant('osmesa', default=False,
+            description='Enable OSMesa support (for VTK)')
+
+    extends('python@3.7:')
+    depends_on('cmake@3.14:', type='build')
+    depends_on('swig@3.0.12', type='build')
 
-    extends('python@2.7.16')
-    depends_on('cmake@3.14:')
-    depends_on('swig@3.0.12')
     depends_on('py-cython')
     depends_on('py-numpy')
-    depends_on('py-pip')
 
     depends_on('eigen@3.3.7')
-    depends_on('cgal@4.13 +shared~core~demos~imageio')
+    depends_on('cgal@4.13 build_type=Release +shared~core~demos~imageio')
```

Comment:
It can't be built with any other type, or you just prefer the Release build?

Reply:
this is a preference. if i remember correctly, cgal default is not Release. but for this purpose, we can remove it if spack is following a standard of not putting build types in..

Reply:
The problem with putting preferences in `depends_on` is that they're hard requirements, so if someone already has `cgal build_type=RelWithDebInfo` built, the `memsurfer` package can't reuse it. Preferences belong in `packages.yaml`.

Reply:
fair point. let's go ahead and do that. i will push that change in.